File: AndroidOCourse/ListViewDemoJava/app/src/main/java/com/kdotz/listviewdemojava/MainActivity.java
package com.kdotz.listviewdemojava;

import android.annotation.TargetApi;
import android.os.Build;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class MainActivity extends AppCompatActivity {

    @TargetApi(Build.VERSION_CODES.N)
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ListView myListView = findViewById(R.id.myListView);
        List<String> myFriends = Stream.of("Karina", "Meghana", "Liz", "Lily", "Melinda", "Lauren", "Franny").collect(Collectors.toList());

        ArrayAdapter<String> arrayAdapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, myFriends);

        myListView.setAdapter(arrayAdapter);

        myListView.setOnItemClickListener((parent, view, position, id) -> {
            Toast.makeText(this, "My friend: " + myFriends.get(position), Toast.LENGTH_SHORT).show();
        });
    }
}

File: AndroidOCourse/ListViewDemo/app/src/main/java/com/kdotz/listviewdemo/MainActivity.kt
package com.kdotz.listviewdemo

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val myListView = findViewById<ListView>(R.id.myListView)
        val myFamily = listOf("Tom", "Lynda", "Jerry", "Vi", "Nick", "Drew")

        val arrayAdapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, myFamily)

        myListView.adapter = arrayAdapter

        myListView.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
            Toast.makeText(this, "My family: " + myFamily[position], Toast.LENGTH_SHORT).show()
        }
    }
}


